google_analytics/ga_homepage_report.py

```diff
@@ -1,365 +1,534 @@
+    program_course_enrolls_df = enrollmentDataframe(program_course_enrolls, 'program', 'Course')
+    course_program_enrolls_df = enrollmentDataframe(course_program_enrolls, 'course', 'Program')
+    course_course_enrolls_df = enrollmentDataframe(course_course_enrolls, 'course', 'Course')
+
+    program_clicks_df = clicksDataframe(program_clicks)
+    course_clicks_df = clicksDataframe(course_clicks)
+
+    program_enrolls_df = mergeEnrollmentByTypeDataframe(program_program_enrolls_df, program_course_enrolls_df, program_clicks_df)
+    course_enrolls_df = mergeEnrollmentByTypeDataframe(course_program_enrolls_df, course_course_enrolls_df, course_clicks_df)
+
+    return (
+        program_enrolls_df,
+        course_enrolls_df
+    )
+
+
+def clicksDataframe(clicks_data):
+    clicks_dataframe = DataFrame(clicks_data, columns=['date', 'cardName', 'position', 'totalClicks', 'uniqueClicks'])
+    clicks_dataframe = clicks_dataframe.convert_objects(convert_numeric=True)
+    clicks_dataframe.drop('date', axis=1, inplace=True)
+
+    return clicks_dataframe
+
+
+def enrollmentDataframe(enrolls_data, card_type, enroll_type):
+    enrolls_dataframe = DataFrame(
+        enrolls_data,
+        columns=[
+            'date',
+            'cardName',
+            'position',
+            'total{type}Enrolls'.format(type=enroll_type),
+            'unique{type}Enrolls'.format(type=enroll_type)
+        ]
+    )
+
+    enrolls_dataframe = enrolls_dataframe.convert_objects(convert_numeric=True)
+    enrolls_dataframe.drop('date', axis=1, inplace=True)
+    enrolls_dataframe['type'] = card_type
+
+    return enrolls_dataframe
+
+
+def mergeEnrollmentByTypeDataframe(program_enrolls, course_enrolls, clicks):
+    dataframe = pandas.merge(
+        program_enrolls,
+        course_enrolls,
+        on=['cardName', 'position', 'type'],
+        how='left'
+    )
+
+    dataframe = pandas.merge(
+        dataframe,
+        clicks,
+        on=['cardName', 'position'],
+        how='left'
+    )
+
+    total_clicks_by_type = dataframe['uniqueClicks'].sum()
+
+    dataframe['uniqueEnrolls'] = dataframe['uniqueCourseEnrolls'] + dataframe['uniqueProgramEnrolls']
+    dataframe['conversionRate'] = dataframe['uniqueEnrolls'] / dataframe['uniqueClicks']
+    dataframe['programConversionRate'] = dataframe['uniqueProgramEnrolls'] / dataframe['uniqueClicks']
+    dataframe['courseConversionRate'] = dataframe['uniqueCourseEnrolls'] / dataframe['uniqueClicks']
+    dataframe['clickShareByType'] = dataframe['uniqueClicks'] / total_clicks_by_type
+    dataframe['enrollsPerClickByType'] = dataframe['uniqueEnrolls'] / total_clicks_by_type
+
+    return dataframe
+
+
+def strip_edx_page_title(page_title):
+    return page_title.replace(' | edX', '')
+
+
+def output_report(filename, total_homepage_views=None, total_course_card_clicks=None, total_program_card_clicks=None, featured_cards=None, homepage_subjects=None):
```

Comment:
This is a script run by marketing which is being sunset.  I'd prefer to keep it all together.